### ISSUE TITLE
Deep-copied graphs have incorrect adjacency tables

### ISSUE THREAD

**Issue by merkys** [graphviz-perl/Graph#25]
MWE:
```Perl
use Graph::Undirected;

my $orig = Graph::Undirected->new;
$orig->add_edge( { name => 'A' }, { name => 'B' } );

my $copy = $orig->deep_copy;

for my $g ($orig, $copy) {
    for ($g->vertices) {
        print scalar $g->neighbours( $_ );
    }
    print "\n";
}
```
Expected output would be
```
11
11
```
meaning that both graphs have two vertices with one neighbour each. However, I get
```
11
00
```
meaning that something is wrong with adjacency table in deep-copied graph. The same result is acquired when copying a graph with `clone()` from `Clone`.

I am using Graph v0.9722. Same results with `Storable` 2.62 and 3.25, `Data::Dumper` 2.167 and 2.183.

**Comment by mohawk2** [comment]
Thanks for the report! On digging into this, I realise that your code isn't using `refvertexed`. I'm making it throw an exception if vertices are refs and non-`refvertexed`, but if `refvertexed` to deep-copy the vertices and recreate edges.

**Comment by mohawk2** [comment]
In fact, while one is "supposed" to use the `refvertexed` flag for that, there's no reason to throw an exception in this situation, so it doesn't. I have made it recreate the vertex map on `deep_copy` if any vertices are refs, and released a new version 0.9723. Please report any problems!

**Comment by merkys** [comment]
Thanks a lot for prompt fix. I confirm that the MWE works as expected with Graph 0.9723. I admit I was not aware of `refvertexed` - I should have read the documentation before coding. Nevertheless I agree that recreating the vertex map if any vertices are refs is much more user-friendly. Thanks once more!
